hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1597,10 +1597,9 @@ private static Map<String, Schema> getColumnsToIndexWithoutRequiredMetaFields(Ho
       tableSchemaLazyOpt.get().map(schema -> getFirstNSupportedFields(schema, metadataConfig.maxColumnsToIndexForColStats(), recordType)).orElse(Stream.empty())
           .forEach(entry -> colsToIndexSchemaMap.put(entry.getKey(), entry.getValue()));
       return colsToIndexSchemaMap;
-    } else if (isTableInitializing) {
-      return Collections.emptyMap();
     } else {
-      throw new HoodieMetadataException("Cannot initialize col stats with empty list of cols");
+      // initialize col stats index config with empty list of cols
+      return Collections.emptyMap();
```

Comment:
can we add tests for this case

Reply:
I don't know how to add it bcause the input write schema is empty too, not a real production use case.

Reply:
Filed HUDI-9194 to follow up as the test to be added does not seem to be important.